src/contracts/core/DelegationManager.sol

```diff
@@ -563,6 +600,11 @@ contract DelegationManager is
         // Remove `withdrawalRoot` from pending roots
         delete pendingWithdrawals[withdrawalRoot];
 
+        require(
+            !isPendingHandoff(withdrawal.delegatedTo) || receiveAsTokens,
+            "DelegationManager._completeQueuedWithdrawal: a staker pending handoff cannot withdraw as shares"
+        );
```

Comment:
feels like a potential major UX headache? Very confusing UX for stakers that get hit by this; users could be forced to withdraw from system when e.g. trying to change delegation.
I'm presuming this behavior is to simplify accounting? what's the alternative solution?